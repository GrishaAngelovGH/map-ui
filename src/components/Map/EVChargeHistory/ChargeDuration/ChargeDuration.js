const ChargeDuration = ({ duration }) => (
  <div className='row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow'>
    <div className='col-md-12'>
      <div className='row'>
        <div className='col-md-2'>
          <i className='bi bi-lightning-charge fs-1'></i>
        </div>
        <div className='col-md-7'>
          <p className='m-0 fw-bold'>{duration.name}</p>
          <p className='m-0 text-secondary'>{duration.location}</p>
        </div>
        <div className='col-md-3 d-flex align-items-center'>
          <i className='bi bi-clock fs-2'></i>
          <span className='fw-bold mx-1'>{duration.durationTimeInMin}m</span>
        </div>
      </div>
      <div className='row justify-content-center'>
        <div className='col-md-4 d-flex align-items-center'>
          <i className='bi bi-battery-half text-danger fs-1'></i>
          <span className='fw-bold text-danger mx-1'>{duration.prevBatteryCharge}%</span>
          <i className='bi bi-arrow-right-short fs-1'></i>
        </div>
        <div className='col-md-4 d-flex align-items-center'>
          <i className='bi bi-battery-half text-success fs-1'></i>
          <span className='fw-bold text-success mx-1'>{duration.currentBatteryCharge}%</span>
        </div>
      </div>
    </div>
  </div>
)

export default ChargeDuration