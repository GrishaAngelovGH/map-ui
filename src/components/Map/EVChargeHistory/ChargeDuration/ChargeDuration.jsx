const ChargeDuration = ({ duration }) => (
  <div className='row mb-4 justify-content-evenly align-items-center border rounded p-2 shadow'>
    <div className='col-md-2'>
      <i className='bi bi-lightning-charge fs-1'></i>
    </div>
    <div className='col-md-6'>
      <p className='m-0 mt-3 fw-bold'>{duration.name}</p>
      <p className='m-0 text-secondary'>{duration.address}</p>

      <div className='d-flex'>
        <div className='w-50 d-flex align-items-center'>
          <i className='bi bi-battery-half text-danger fs-2 d-flex'></i>
          <span className='fw-bold text-danger mx-1'>{duration.prevBatteryCharge}%</span>
        </div>
        <div className='w-50 d-flex align-items-center'>
          <i className='bi bi-battery-half text-success fs-2 d-flex'></i>
          <span className='fw-bold text-success mx-1'>{duration.currentBatteryCharge}%</span>
        </div>
      </div>
    </div>
    <div className='col-md-3 d-flex align-items-center'>
      <i className='bi bi-clock fs-1'></i>
      <span className='fw-bold mx-2'>{duration.durationTimeInMin}m</span>
    </div>
  </div>
)

export default ChargeDuration