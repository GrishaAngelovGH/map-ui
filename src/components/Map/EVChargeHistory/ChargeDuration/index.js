import ChargeDuration from './ChargeDuration'

export default ChargeDuration