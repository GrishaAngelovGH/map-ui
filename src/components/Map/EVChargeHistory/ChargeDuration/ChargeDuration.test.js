import { render } from '@testing-library/react'

import ChargeDuration from './ChargeDuration'

test('should render ChargeDuration component', () => {
  const view = render(
    <ChargeDuration
      duration={{
        location: { city: 'London', country: 'England, UK' },
        currentBatteryCharge: 85,
        prevBatteryCharge: 70,
        durationTimeInMin: 35,
        charge: true,
        name: 'ChargePoint - 123 Road',
        location: 'A12 B34 London, UK'
      }}
    />
  )

  expect(view).toMatchSnapshot()
})