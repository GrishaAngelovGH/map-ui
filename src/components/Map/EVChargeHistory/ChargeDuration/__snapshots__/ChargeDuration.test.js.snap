// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChargeDuration component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row"
          >
            <div
              class="col-md-2"
            >
              <i
                class="bi bi-lightning-charge fs-1"
              />
            </div>
            <div
              class="col-md-7"
            >
              <p
                class="m-0 fw-bold"
              >
                ChargePoint - 123 Road
              </p>
              <p
                class="m-0 text-secondary"
              >
                A12 B34 London, UK
              </p>
            </div>
            <div
              class="col-md-3 d-flex align-items-center"
            >
              <i
                class="bi bi-clock fs-2"
              />
              <span
                class="fw-bold mx-1"
              >
                35
                m
              </span>
            </div>
          </div>
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-4 d-flex align-items-center"
            >
              <i
                class="bi bi-battery-half text-danger fs-1"
              />
              <span
                class="fw-bold text-danger mx-1"
              >
                70
                %
              </span>
              <i
                class="bi bi-arrow-right-short fs-1"
              />
            </div>
            <div
              class="col-md-4 d-flex align-items-center"
            >
              <i
                class="bi bi-battery-half text-success fs-1"
              />
              <span
                class="fw-bold text-success mx-1"
              >
                85
                %
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
    >
      <div
        class="col-md-12"
      >
        <div
          class="row"
        >
          <div
            class="col-md-2"
          >
            <i
              class="bi bi-lightning-charge fs-1"
            />
          </div>
          <div
            class="col-md-7"
          >
            <p
              class="m-0 fw-bold"
            >
              ChargePoint - 123 Road
            </p>
            <p
              class="m-0 text-secondary"
            >
              A12 B34 London, UK
            </p>
          </div>
          <div
            class="col-md-3 d-flex align-items-center"
          >
            <i
              class="bi bi-clock fs-2"
            />
            <span
              class="fw-bold mx-1"
            >
              35
              m
            </span>
          </div>
        </div>
        <div
          class="row justify-content-center"
        >
          <div
            class="col-md-4 d-flex align-items-center"
          >
            <i
              class="bi bi-battery-half text-danger fs-1"
            />
            <span
              class="fw-bold text-danger mx-1"
            >
              70
              %
            </span>
            <i
              class="bi bi-arrow-right-short fs-1"
            />
          </div>
          <div
            class="col-md-4 d-flex align-items-center"
          >
            <i
              class="bi bi-battery-half text-success fs-1"
            />
            <span
              class="fw-bold text-success mx-1"
            >
              85
              %
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
